src/components/CardList/CardList.tsx

```diff
@@ -0,0 +1,48 @@
+import { MouseEvent, useContext } from 'react';
+
+import CardInfoListContext from '../../contexts/CardInfoContext';
+
+import { CardInfoList } from '../../types/state';
+import { PATHNAME } from '../../constants/pathname';
+import { generateCardKey } from '../../domains/keyGenerator';
+import { useInitCardInfo } from '../../hooks/useInitCardInfo';
+import { useNavigationTo } from '../../hooks/useNavigationTo';
+
+import * as styled from './CardList.styled';
+import Card from '../Card/Card';
+
+const CardList = () => {
+  const { cardInfoList } = useContext(CardInfoListContext);
+  const { navigationTo } = useNavigationTo(PATHNAME.CARD_ALIAS);
+  const initCardInfo = useInitCardInfo();
+
+  const onClick = ({ currentTarget }: MouseEvent<HTMLDivElement>) => {
+    if (!currentTarget.dataset.cardInfo) {
+      return;
+    }
+
+    const cardInfo = JSON.parse(currentTarget.dataset.cardInfo);
+    initCardInfo(cardInfo);
+    navigationTo();
+  };
+
+  const generateCardList = (cardInfoList: CardInfoList) => {
+    return Object.values(cardInfoList).map(cardInfo => {
+      const key = generateCardKey(cardInfo);
+      const data = JSON.stringify(cardInfo);
+
+      return (
+        <styled.CardContainer key={key + 'container'} onClick={onClick} data-card-info={data}>
+          <Card key={key + 'card'} cardInfo={cardInfo} theme={cardInfo.cardCompany.theme} />
+          {cardInfo.cardAlias && (
+            <styled.CardAlias key={key + 'alias'}>{cardInfo.cardAlias}</styled.CardAlias>
+          )}
+        </styled.CardContainer>
+      );
+    });
+  };
+
+  return <>{cardInfoList && <styled.CardList>{generateCardList(cardInfoList)}</styled.CardList>}</>;
+};
```

Comment:
조건문은 항상 읽는데 부담이 되기때문에 early return 을 적용해서 예외상황은 상단에서 처리하고 정상케이스만 나눠서 보자는 의미였어요.

&& 연산자로 버그가 발생할수있다는 부분은 어떤 내용인지 공유해주실수있을까요?

Reply:
[리액트 - 조건부 렌더링](https://react.dev/learn/conditional-rendering)

Pitfall
Don’t put numbers on the left side of &&.

To test the condition, JavaScript converts the left side to a boolean automatically. However, if the left side is 0, then the whole expression gets that value (0), and React will happily render 0 rather than nothing.

For example, a common mistake is to write code like messageCount && <p>New messages</p>. It’s easy to assume that it renders nothing when messageCount is 0, but it really renders the 0 itself!

To fix it, make the left side a boolean: messageCount > 0 && <p>New messages</p>.

리액트 docs에서 이야기하는 해당 부분입니다!
&& 연산자의 왼쪽 피연산자의 값이 0일 경우 0이 렌더링될 수 있는 UI 버그에 대해 말씀드린 것이었습니다~!

Reply:
아하 falsy/truthy 비교는 지양해보면 좋겠네요~